sanic/headers.py

```diff
@@ -33,173 +33,237 @@
 # For more information, consult ../tests/test_requests.py
 
 
-def parse_arg_as_accept(f):
-    def func(self, other, *args, **kwargs):
-        if not isinstance(other, Accept) and other:
-            other = Accept.parse(other)
-        return f(self, other, *args, **kwargs)
+class MediaType:
+    """A media type, as used in the Accept header."""
 
-    return func
-
-
-class MediaType(str):
-    def __new__(cls, value: str):
-        return str.__new__(cls, value)
+    def __init__(
+        self,
+        type_: str,
+        subtype: str,
+        **params: str,
+    ):
+        self.type = type_
+        self.subtype = subtype
+        self.q = float(params.get("q", "1.0"))
+        self.params = params
+        self.mime = f"{type_}/{subtype}"
+        self.key = (
+            -1 * self.q,
+            -1 * len(self.params),
```

Comment:
Could use simple negation rather than multiplication (I think Python optimizes but for human readers...)